src/remotestorage.js

```diff
@@ -1,829 +1,792 @@
-     *
-     */
-    getRequestTimeout: function () {
-      return config.requestTimeout;
-    },
-
-    syncCycle: function () {
-      if (this.sync.stopped) {
-        return;
+    }
+    this._setGPD({
+      get: this._pendingGPD('get'),
+      put: this._pendingGPD('put'),
+      delete: this._pendingGPD('delete')
+    });
+    var n = this._cleanups.length, i = 0;
+
+    var oneDone = function () {
+      i++;
+      if (i >= n) {
+        this._init();
+        log('Done cleaning up, emitting disconnected and disconnect events');
+        this._emit('disconnected');
       }
+    }.bind(this);
 
-      this.sync.on('done', function () {
-        log('[Sync] Sync done. Setting timer to', this.getCurrentSyncInterval());
-        if (!this.sync.stopped) {
-          if (this._syncTimer) {
-            clearTimeout(this._syncTimer);
-          }
-          this._syncTimer = setTimeout(this.sync.sync.bind(this.sync), this.getCurrentSyncInterval());
+    if (n > 0) {
+      this._cleanups.forEach(function (cleanup) {
+        var cleanupResult = cleanup(this);
+        if (typeof(cleanupResult) === 'object' && typeof(cleanupResult.then) === 'function') {
+          cleanupResult.then(oneDone);
+        } else {
+          oneDone();
         }
       }.bind(this));
+    } else {
+      oneDone();
+    }
+  },
 
-      this.sync.sync();
-    },
-
-    stopSync: function () {
-      if (this.sync) {
-        log('[Sync] Stopping sync');
-        this.sync.stopped = true;
+  /**
+   * TODO: document
+   *
+   * @private
+   */
+  setBackend: function (what) {
+    this.backend = what;
+    if (hasLocalStorage) {
+      if (what) {
+        localStorage.setItem('remotestorage:backend', what);
       } else {
-        // TODO When is this ever the case and what is syncStopped for then?
-        log('[Sync] Will instantiate sync stopped');
-        this.syncStopped = true;
+        localStorage.removeItem('remotestorage:backend');
       }
-    },
+    }
+  },
 
-    startSync: function () {
-      if (!config.cache) { return; }
-      this.sync.stopped = false;
-      this.syncStopped = false;
-      this.sync.sync();
+  /**
+   * Add a "change" event handler to the given path. Whenever a "change"
+   * happens (as determined by the backend, such as e.g.
+   * <RemoteStorage.IndexedDB>) and the affected path is equal to or below the
+   * given 'path', the given handler is called.
+   *
+   * You should usually not use this method directly, but instead use the
+   * "change" events provided by <RemoteStorage.BaseClient>.
+   *
+   * @param path    - Absolute path to attach handler to
+   * @param handler - Handler function
+   */
+  onChange: function (path, handler) {
+    if (! this._pathHandlers.change[path]) {
+      this._pathHandlers.change[path] = [];
     }
+    this._pathHandlers.change[path].push(handler);
+  },
 
-  };
+  /**
+   * TODO: do we still need this, now that we always instantiate the prototype?
```

Comment:
Not sure I get the question, but I think we still need this for enabling/disabling the logging during runtime (i.e. after initial creation of the remoteStorage instance).

Reply:
Yes, but why would you do that? Or, in other words, did you ever do that yourself? And if so, did you not just do it, because you couldn't instantiate the instance?

Reply:
I can't really anticipate all use-cases others might have. I myself used it a few times for debugging, enabling the log for a certain action and then disabling it again so it's not so noisy.

Reply:
It's not really about anticipating, but more like giving away clothes you haven't used in the last 12 months. Enabling the logs for a certain action and then disabling before reload seems like a rather rare use case to me.

However, adding this function instead of the init config, and then losing lots of logs without realizing it (which happened to me every time I used it myself) seems like an actual problem we should try to not have people run into.